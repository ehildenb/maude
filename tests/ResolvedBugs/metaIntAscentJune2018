#!/bin/sh

MAUDE_LIB=$srcdir/../../src/Main
export MAUDE_LIB

../../src/Main/maude \
  < $srcdir/metaIntAscentJune2018.maude -no-banner -no-advise \
  > metaIntAscentJune2018.out 2>&1

diff $srcdir/metaIntAscentJune2018.expected metaIntAscentJune2018.out > /dev/null 2>&1
