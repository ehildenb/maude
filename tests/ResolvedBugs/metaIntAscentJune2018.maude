set show timing off .
set show advisories off .

***
***	Check that the meta-interpreter ascent functions get module imports from from the
***	meta-interpreter and not the object level interpreter.
***

set include BOOL off .

fmod FOO is
  sort ObjectLevelFoo .
endfm

load metaInterpreter 

mod TEST is
  pr META-INTERPRETER .
  op me : -> Oid .
  op User : -> Cid .
  op state:_ : Nat -> Attribute .

  op foo : -> Module .
  eq foo = (
fmod 'FOO is
  nil
  sorts 'MetaLevelFoo .
  none
  none
  none
  none
endfm) .

  vars X Y Z : Oid .

  rl < X : User | state: 0 > createdInterpreter(X, Y, Z) =>
     < X : User | state: 1 > insertModule(Z, X, foo) .

  rl < X : User | state: 1 > insertedModule(X, Y) =>
     < X : User | state: 2 > insertModule(Y, X, upModule('META-LEVEL, true)) .

  rl < X : User | state: 2 > insertedModule(X, Y) =>
     < X : User | state: 3 > reduceTerm(Y, X, 'META-LEVEL,
        'upModule[''FOO.Qid, 'true.Bool]) .

  op foo : -> Module .
  eq foo = (
fmod 'FOO is
  nil
  sorts 'MetaLevelFoo .
  none
  none
  none
  none
endfm) .

endm

*** expecting meta-meta-module to contain sort MetaLevelFoo but get ObjectLevelFoo
erew  <> < me : User | state: 0 > createInterpreter(interpreterManager, me, none) .
