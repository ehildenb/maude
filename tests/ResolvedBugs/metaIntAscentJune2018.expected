==========================================
erewrite in TEST : <> createInterpreter(interpreterManager, me) < me : User |
    state: 0 > .
rewrites: 7
result Configuration: <> < me : User | state: 3 > reducedTerm(me, interpreter(
    0), 1, 'fmod_is_sorts_.____endfm[''FOO.Sort,'nil.ImportList,
    ''MetaLevelFoo.Sort,'none.SubsortDeclSet,'none.OpDeclSet,'none.MembAxSet,
    'none.EquationSet], 'FModule)
Bye.
